base/pkg/entry.jl

```diff
@@ -735,13 +736,32 @@ function build!(pkgs::Vector, errs::Dict, seen::Set=Set())
         isfile(path) || continue
         info("Building $pkg")
         cd(dirname(path)) do
-            try evalfile(path)
-            catch err
+            code = """
+               try
+                   include("$(escape_string(path))")
+               catch e
+                   open("$(escape_string(errfile))", "w") do f
+                       serialize(f, e)
+                   end
+                   exit(1)
+               end
+            """
+            isfile(errfile) && Base.rm(errfile)
+            try
+                run(`$(julia_cmd()) --startup-file=no --history-file=no
+                     --color=$(Base.have_color ? "yes" : "no") --eval $code`)
+            catch e
+                err = try
+                    open(deserialize, errfile, "r")
+                catch
+                    e
+                end
                 warnbanner(err, label="[ ERROR: $pkg ]")
                 errs[pkg] = err
             end
         end
     end
+    isfile(errfile) && Base.rm(errfile)
```

Comment:
Maybe, but that seems like a separate PR.  We aren't keeping a log file now.

The temporary file used here is poorly suited as a log file because I'm re-using the same temporary file for building each package.

Further, the only information in the temporary file is a serialized exception, which is already deserialized and stored in the `errs` array, so I don't understand what would be gained from keeping the file around.